tests/semantic_analyser.cpp

```diff
@@ -2832,6 +2832,12 @@ TEST(semantic_analyser, binop_sign)
   }
 }
 
+TEST(semantic_analyser, binop_late_ptr_resolution)
+{
+  test("kfunc:mem_cgroup_charge_skmem { if (@a[1] == args.memcg) { print((1)); "
```

Comment:
not sure if we should use a real fentry function in this test but I'm not sure how else to test. Maybe with a uprobe and our own program?

Reply:
Use `semantic_analyser_btf` fixture and test on functions from `tests/data/data_source.c`. Also prefer `fentry` over `kfunc`.

Reply:
Thanks @viktormalik - that works.